python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -0,0 +1,289 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Translate polars IR representation to ours."""
+
+from __future__ import annotations
+
+from contextlib import AbstractContextManager, nullcontext
+from typing import Any
+
+from polars.polars import _expr_nodes as pl_expr, _ir_nodes as pl_ir
+
+from cudf_polars.dsl import expr, ir
+from cudf_polars.utils import dtypes
+
+__all__ = ["translate_ir", "translate_expr"]
+
+
+class set_node(AbstractContextManager):
+    """Run a block with current node set in the visitor."""
+
+    __slots__ = ("n", "visitor")
```

Comment:
Should this be `old_n` to clarify that it's backing up the prior value to reset after the current visit completes?

Reply:
Oh it didn't fully register what was happening in init vs enter, it's basically a backup location used for swapping. Yeah I guess the most self-documenting solution would be not to reuse the slot but to have separate ones for the user-provided node and the one that's being backed up when the CM enters, but that is extra memory usage. I think renaming to node is sufficient, thanks.